### PR TITLE
antlr3: add formula for antlr3.4

### PR DESCRIPTION
**Pull request by bagage**
Antlr4 is the current main, but some people may need Antlr3, which has support for a C API.

### CODE REVIEW

File: antlr3.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Antlr3 < Formula
+  # Antlr4 is the current main, but some people may need Antlr3,
+  # which has support for a C API.
+  url "http://www.antlr3.org/download/antlr-3.4-complete.jar"
+  version '3.4'
```
Comment (DomT4) [inline antlr3.rb]: Double quotes everywhere.

File: antlr3.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  homepage 'http://www.antlr.org/'
+  sha1 '5cab59d859caa6598e28131d30dd2e89806db57f'
```
Comment (DomT4) [inline antlr3.rb]: `sha256` here please.

File: antlr3.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  def install
+    prefix.install "antlr-3.4-complete.jar"
```
Comment (DomT4) [inline antlr3.rb]: Would prefer to put this in `libexec` and fix the other `prefix` calls below.

File: antlr3.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    (share+"java").install_symlink "#{prefix}/antlr-3.4-complete.jar" => "antlr3.jar"
+    (bin+"antlr-3.4").write <<-EOS.undent
+    #!/bin/sh
```
Comment (DomT4) [inline antlr3.rb]: Indent this two spaces so it's obvious it's not part of the main formula.

**Comment by bagage** [comment]
Done - I'm not totally sure for `libexec` usage; is it ok now?

**Comment by DomT4** [comment]
Changes look good, Thanks. Our test-bot flagged a few remaining issues:

```
==> audit problems
homebrew/versions/antlr3:
 * `homepage`(line 6) should be put before `url`(line 4)
 * A `test do` test block should be added
 * Stable: version 3.4 is redundant with version scanned from URL
 * Formula should have a desc (Description).
```

**Comment by bagage** [comment]
Updated - I did not do a test: section though since I do not know what to test.

**Comment by DomT4** [comment]
Presuming it works, you could steal the test from [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/antlr.rb#L16-L34).

**Comment by DomT4** [comment]
Ping?

**Comment by bagage** [comment]
My bad I totally forgot this. I tried to c/c the test but antlr 3.4 does not generate anything using the method of antlr4. I did not investigate why though - does this really matter or can we keep it without test?

**Comment by DomT4** [comment]
Yeah, I can see why now I've looked at it. One of the tools used in that test looks like a new introduction to Antlr4.

There are some [examples](https://stackoverflow.com/questions/1931307/antlr-is-there-a-simple-example) at that link, It'd be nice to try and adapt them for this PR. It's troublesome to accept things without tests because we don't have a way of knowing the formula actually works at runtime without them.

**Comment by bagage** [comment]
Finally done!

**Comment by dunn** [comment]
Just one little thing left:
```
==> audit problems
homebrew/versions/antlr3:
 * Description is too long. "name: desc" should be less than 80 characters.
Length is calculated as homebrew/versions/antlr3 + desc. (currently 124)
```

**Comment by bagage** [comment]
Done again? This time everything is green! :)

File: antlr3.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+class Antlr3 < Formula
+  desc "Antlr3 has a C API which has been dropped in Antlr4."
+  homepage "http://www.antlr.org/"
+  url "http://www.antlr3.org/download/antlr-3.4-complete.jar"
+  sha256 "9d3e866b610460664522520f73b81777b5626fb0a282a5952b9800b751550bf7"
+
+  def install
+    libexec.install "antlr-3.4-complete.jar"
+    (share+"java").install_symlink "#{libexec}/antlr-3.4-complete.jar" => "antlr3.jar"
+    (bin+"antlr-3.4").write <<-EOS.undent
```
Comment (dunn) [inline antlr3.rb]: Why not just `(bin/"antlr3").write`, and skip line 14 below? Is there a reason to have the executable available under two different names?

Comment (bagage) [inline antlr3.rb]: indeed, fixed.

**Comment by bagage** [comment]
Any news?

File: antlr3.rb (hunk 5 above)
Comment (dunn) [inline antlr3.rb]: Do we want `bin/antlr3` or `bin/antlr-3.4`?

File: antlr3.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      #!/bin/sh
+      java -jar #{libexec}/antlr-3.4-complete.jar "$@"
+      EOS
```
Comment (DomT4) [inline antlr3.rb]: `EOS` should line up with the start of `(bin+"antlr-3.4")`.write`

File: antlr3.rb (hunk 5 above)
Comment (bagage) [inline antlr3.rb]: I think `bin/antlr-3.4` is better since `antlr-3.2` exists too.

File: antlr3.rb (hunk 6 above)
Comment (bagage) [inline antlr3.rb]: aligned.

File: antlr3.rb (hunk 5 above)
Comment (dunn) [inline antlr3.rb]: We don't have antlr 3.2 in this tap, though.  We do have antlr 2, and it's installed as `bin/antlr2`.

Comment (dunn) [inline antlr3.rb]: If you think we should have 3.2, then maybe `antlr34`?

Comment (bagage) [inline antlr3.rb]: Well if antlr2 is named as it, let's name it `antlr3` (`antlr3.2` is broken anyway, so nobody should use it!). Fixed.

**Comment by bagage** [comment]
Bump?

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
@bagage Apologies, busy few days. If all is green on the test-bot this time as well I'll merge it shortly after.

**Comment by bagage** [comment]
Sure, no prob. Thanks!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-versions/commit/857638c6fbb7ab9b2d322d2dee4308f101647ce6. Thank you for your contribution to Versions, we appreciate it! :smiley_cat:

**Pull request closed without merging**